### DISCUSSION TITLE
How to deal with special keys

### DISCUSSION THREAD

**Discussion by artuntun** [pydantic/pydantic#4176] [Question]
Hey there

I am trying to initialize a BaseModel from a dictionary. The dictionary looks something like this:

```
data = {
    key-1: value
    class: value2
}
```

The model I wrote:

```python
class RawData(BaseModel):
    key_1: str
    class: str
```

There are two issues here. First the class method can't be used since is a protected name and secondly, I can't use the _-_
 as in `key-1` for naming an attribute. I ended up doing a preprocessing with:
 
```python
class RawData(BaseModel):
    key_1: str
    token: str
def preprocess(data: Dict[str, str]):
    data['key_1'] = data['key-1']
    del data['key-1]
    data['label'] = data['class']
    del data['class']

preprocess(data)
raw_data = RawData(**data)
 ```

Is there any way to add this preprocessing into the RawData model?

Thank you

**Comment by samuelcolvin** [comment]
You need `alias`. Check the docs.

**Reply by artuntun** [reply]
thanks!
